lib/avromatic/model/validation.rb

```diff
@@ -0,0 +1,48 @@
+module Avromatic
+  module Model
+    module Validation
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+
+        # Returns an array of messages
+        def validate_nested_value(value)
+          case value
+          when Avromatic::Model::Attributes
+            validate_record_value(value)
+          when Array
+            value.map.with_index do |element, index|
+              validate_nested_value(element).map do |message|
+                "[#{index}]#{message}"
+              end
+            end.flatten
+          when Hash
+            value.map do |key, map_value|
```

Comment:
Coercion for hash keys should happen due to the attribute type used with Virtus: `Hash[String => ???]`.